zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -504,60 +512,97 @@ impl<F: Field> SignVerifyChip<F> {
 
         let pk_le = pk_bytes_le(&sign_data.pk);
         let pk_be = pk_bytes_swap_endianness(&pk_le);
-        let pk_hash = (!padding)
-            .then(|| keccak256(&pk_be))
-            .unwrap_or_default()
-            .map(|byte| Value::known(F::from(byte as u64)));
-        let pk_hash_hi = pk_hash[..12].to_vec();
+        let mut pk_hash = (!padding).then(|| keccak256(&pk_be)).unwrap_or_default();
+        pk_hash.reverse();
+
+        let powers_of_256 = iter::successors(Some(F::ONE), |coeff| Some(F::from(256) * coeff))
+            .take(20)
```

Comment:
Here take(16) should be sufficient and more precise for the usage below (since now we split lo/hi)

Reply:
fixed in [24c67ee](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1418/commits/24c67eea3617aaa417a24fd5e1a0f21895fe06f7)